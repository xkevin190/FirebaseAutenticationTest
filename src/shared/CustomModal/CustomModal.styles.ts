import {StyleSheet, ViewStyle} from 'react-native';

import {COLORS} from '../../constants/styles';

const styles = StyleSheet.create({
  modalBackgroundContainer: {
    backgroundColor: COLORS.BLACK_025,
    width: '100%',
    flex: 1,
    justifyContent: 'flex-end',
  } as ViewStyle,
  modalContainer: {
    backgroundColor: COLORS.WHITE,
    width: '100%',
    height: '90%',
    borderTopLeftRadius: 13,
    borderTopRightRadius: 13,
  } as ViewStyle,
});

export default styles;
