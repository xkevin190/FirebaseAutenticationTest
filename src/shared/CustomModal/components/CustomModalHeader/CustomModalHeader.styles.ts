import { StyleSheet, ViewStyle } from 'react-native';

const styles = StyleSheet.create({
  modalHeaderContainer: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 30,
    marginTop: 25,
    paddingHorizontal: 25,
  } as ViewStyle,
  closeButtonContainer: {
    height: '100%',
    width: '100%',
    position: 'absolute',
    justifyContent: 'center',
  } as ViewStyle,
});

export default styles;
