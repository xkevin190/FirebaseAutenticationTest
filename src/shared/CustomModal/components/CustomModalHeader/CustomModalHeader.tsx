import React from 'react';
import {View} from 'react-native';
import Icon from 'react-native-vector-icons/Fontisto';

import CustomText from '../../../CustomText';
import {ICONS} from '../../../../constants/assets';
import {COLORS} from '../../../../constants/styles';
import styles from './CustomModalHeader.styles';

export interface ICustomModalHeaderProps {
  title: string;
  closeModal: () => void;
}

const CustomModalHeader: React.FC<ICustomModalHeaderProps> = ({
  title,
  closeModal,
}) => (
  <View style={styles.modalHeaderContainer}>
    <View style={styles.closeButtonContainer}>
      <Icon
        name={ICONS.FONTISTO.CLOSE}
        size={13}
        color={COLORS.SECONDARY_LIGHT}
        onPress={closeModal}
      />
    </View>
    <CustomText
      text={title}
      color={COLORS.GRAY_DEFAULT_DARKER}
      bodyType="medium-paragraph"
    />
  </View>
);

export default CustomModalHeader;
