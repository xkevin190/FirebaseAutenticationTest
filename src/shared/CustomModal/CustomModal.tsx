import React from 'react';
import { Modal, KeyboardAvoidingView, View } from 'react-native';

import CustomModalHeader from './components/CustomModalHeader';
import styles from './CustomModal.styles';

export interface ICustomModalProps {
  visible: boolean;
  title: string;
  children: React.ReactNode;
  onRequestClose: () => void;
}

const CustomModal: React.FC<ICustomModalProps> = ({
  visible,
  children,
  title,
  onRequestClose,
}) => (
  <Modal
    visible={visible}
    transparent
    statusBarTranslucent
    animationType="fade"
    onRequestClose={onRequestClose}
  >
    <KeyboardAvoidingView
      behavior="padding"
      style={styles.modalBackgroundContainer}
    >
      <View style={styles.modalContainer}>
        <CustomModalHeader title={title} closeModal={onRequestClose} />
        {children}
      </View>
    </KeyboardAvoidingView>
  </Modal>
);

export default CustomModal;
