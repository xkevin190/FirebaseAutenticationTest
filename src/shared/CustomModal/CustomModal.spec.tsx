import React from 'react';
import { ModalProps } from 'react-native';
import { shallow, ShallowWrapper } from 'enzyme';

import { ICustomModalHeaderProps } from './components/CustomModalHeader/CustomModalHeader';
import CustomModal, { ICustomModalProps } from './CustomModal';

describe('CustomModal (Unit-Test)', () => {
  let wrapper: ShallowWrapper;
  const props: ICustomModalProps = {
    onRequestClose: jest.fn(),
    title: 'TEST_MODAL_TITLE',
    visible: true,
    children: null,
  };

  beforeEach(() => {
    wrapper = shallow(<CustomModal {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render modal', () => {
    const modalProps = wrapper.find('Modal').at(0).props() as ModalProps;
    expect(modalProps.visible).toEqual(props.visible);
    expect(modalProps.statusBarTranslucent).toBeTruthy();
    expect(modalProps.animationType).toEqual('fade');
    expect(modalProps.onRequestClose).toEqual(props.onRequestClose);
  });

  it('should render modal header', () => {
    const modalHeaderProps = wrapper
      .find('CustomModalHeader')
      .at(0)
      .props() as ICustomModalHeaderProps;
    expect(modalHeaderProps.title).toEqual(props.title);
    expect(modalHeaderProps.closeModal).toEqual(props.onRequestClose);
  });
});
