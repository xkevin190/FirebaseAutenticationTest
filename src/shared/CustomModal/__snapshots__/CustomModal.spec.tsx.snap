// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal (Unit-Test) should match snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <KeyboardAvoidingView
    behavior="padding"
    style={
      Object {
        "backgroundColor": "#00000040",
        "flex": 1,
        "justifyContent": "flex-end",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderTopLeftRadius": 13,
          "borderTopRightRadius": 13,
          "height": "90%",
          "width": "100%",
        }
      }
    >
      <CustomModalHeader
        closeModal={[MockFunction]}
        title="TEST_MODAL_TITLE"
      />
    </View>
  </KeyboardAvoidingView>
</Modal>
`;
